test/t8_test_cmesh_copy.c

```diff
@@ -37,40 +47,249 @@ test_cmesh_committed (t8_cmesh_t cmesh)
+}
+
+
+static              t8_cmesh_t
+t8_test_create_comm_only_cmesh(int cmesh_id)
+{
+  switch ((int)(cmesh_id/3)) {
+  case 0:
+    return t8_cmesh_new_periodic_tri (comm[cmesh_id % 3]);
+  case 1:
+    return t8_cmesh_new_periodic_hybrid (comm[cmesh_id % 3]);
+  case 2:
+    return t8_cmesh_new_periodic_line_more_trees (comm[cmesh_id % 3]);
+  case 3:
+    return t8_cmesh_new_line_zigzag (comm[cmesh_id % 3]);
+  case 4:
+    return t8_cmesh_new_prism_deformed (comm[cmesh_id % 3]);
+  case 5:
+    return t8_cmesh_new_prism_cake_funny_oriented (comm[cmesh_id % 3]);
+  case 6:
+    return t8_cmesh_new_prism_geometry (comm[cmesh_id % 3]);
+  case 7:
+    return t8_cmesh_new_tet_orientation_test (comm[cmesh_id % 3]);
+  case 8:
+    return t8_cmesh_new_hybrid_gate (comm[cmesh_id % 3]);
+  case 9:
+    return t8_cmesh_new_hybrid_gate_deformed (comm[cmesh_id % 3]);
+  }
+}
+
+static              t8_cmesh_t
+t8_test_create_new_hypercube_cmesh(int cmesh_id)
+{
+  return t8_cmesh_new_hypercube ((t8_eclass_t) (int)(cmesh_id/(num_comm*binary*binary*binary) %(num_comm*binary*binary*binary)),
+                                 comm[((int)cmesh_id/binary*binary*binary)%binary*binary], (int)((cmesh_id%binary*binary*binary)/binary*binary),
+                                 ((int)(cmesh_id/binary))%binary, cmesh_id % binary);
+}
+
+static              t8_cmesh_t
+t8_test_create_new_empty_cmesh(int cmesh_id)
+{
+  return t8_cmesh_new_empty ((int)((cmesh_id/8)%8), (int)((cmesh_id/(dim+2))%binary), cmesh_id % (dim+2));
+}
+
+static              t8_cmesh_t
+t8_test_create_new_from_class_cmesh(int cmesh_id)
+{
+  return t8_cmesh_new_from_class ((t8_eclass_t)((int) (cmesh_id/3 % T8_ECLASS_COUNT)), comm[cmesh_id % 3]);
+}
+
+static              t8_cmesh_t
+t8_test_create_new_hypercube_hybrid_cmesh(int cmesh_id)
+{
+  return t8_cmesh_new_hypercube_hybrid (((int)((cmesh_id/12)%12))%4, comm[(int)((cmesh_id/3)%3)], ((int)(cmesh_id/2))%2, cmesh_id%2);
+}
+
+
+static              t8_cmesh_t
+t8_test_create_new_periodic_cmesh(int cmesh_id)
+{
+  return t8_cmesh_new_periodic (comm[(int)((cmesh_id/4)%8)], (cmesh_id%dim)+min_dim);
+}
+
+static              t8_cmesh_t
+t8_test_create_new_bigmesh_cmesh(int cmesh_id)
+{
+  return t8_cmesh_new_bigmesh ((t8_eclass_t)((int)(cmesh_id/300)%300), (int) ((cmesh_id/3)%100), comm[cmesh_id%2]);
+}
+
+static              t8_cmesh_t
+t8_test_create_new_prism_cake_cmesh(int cmesh_id)
+{
+  return t8_cmesh_new_prism_cake (comm[(int)((cmesh_id/100)%3)], (cmesh_id% 100)+3);
+}
+
+static              t8_cmesh_t
+t8_test_create_new_disjoint_bricks_cmesh(int cmesh_id)
+{
+  return t8_cmesh_new_disjoint_bricks (t8_gloidx_t num_x, t8_gloidx_t num_y, t8_gloidx_t num_z,int x_periodic,int y_periodic,int z_periodic,sc_MPI_Comm comm);
+}
+
+static              t8_cmesh_t
+t8_test_create_cmesh(int cmesh_id)
+{
+  if(0<=cmesh_id<t8_get_comm_only_cmesh_testcases()){
+    return t8_test_create_comm_only_cmesh(cmesh_id);
+  }
```

Comment:
To a void the evergrowing lines, modify your variable each time and get rid of the else:

```
}
cmesh_id -= t8_get_comm_only_cmesh_testcases();
if (0 <= cmesh_id && cmesh_id < t8_get_new_hypercube_cmesh_testcases()) {
   return t8_test_create_new_hypercube_cmesh(cmesh_id);
}
cmesh_id -= t8_get_comm_only_cmesh_testcases();
if (0 <= ...
```